OpenRA.Game/Settings.cs

```diff
@@ -300,6 +301,28 @@ public void Save()
 			root.WriteToFile(settingsFile);
 		}
 
+		public static string SanitizedPlayerName(string dirty)
+		{
+			// reserved characters for MiniYAML and JSON
+			var disallowedChars = new char[] { ' ', '#', '@', ':', '\n', '\t', '[', ']', '{', '}', '"', '`' };
+			var forbiddenNames = new string[] { "Open", "Closed", "Spectator" };
+
+			var clean = dirty;
+			if (string.IsNullOrEmpty(dirty) || forbiddenNames.Contains(clean) || clean.Contains(" AI"))
```

Comment:
A simpler and more robust solution would be to check against the AI names directly.